Cm/pse 09 (#28)
Ada-Developers-Academy/core-pses: 10/stock.checkpoint.md

from main import max_profit

class TestPython1(unittest.TestCase):
    def test_profit_7_1_5_3_6_4(self):
        # Arrange
        arr = [7, 1, 5, 3, 6, 4]

        # Act
        answer = max_profit(arr)

        # Assert
        self.assertEqual(answer, 7)

    def test_profit_1_2_3_4_5(self):
        # Arrange
        arr = [1, 2, 3, 4, 5]

        # Act
        answer = max_profit(arr)

        # Assert
        self.assertEqual(answer, 4)

    def test_profit_7_6_4_3_1(self):
        # Arrange
        arr = [7, 6, 4, 3, 1]

        # Act
        answer = max_profit(arr)

        # Assert
        self.assertEqual(answer, 0)
```

##### !end-tests

##### !explanation

Our solution:

```python
def max_profit(prices):
    total_profit = 0
    for i in range(1, len(prices)):
        # if prices[i] - prices[i-1] is > 0, then we have a profit
        total_profit += max(prices[i]-prices[i-1], 0)
    return total_profit
```

##### !end-explanation

### !end-challenge

<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->

### !challenge

* type: short-answer
* id: 9707003f-e842-446b-9c39-e90e62a9da85
* title: What is the time complexity of your solution?

> * points: 0

Reviewer: ```suggestion
* points: 1
```

* topics: Big-O, python, lists

##### !question

> What is the time complexity of your answer?

Reviewer: ```suggestion
What is the time complexity of your solution? Explain. Define your variable(s).
```

##### !end-question

##### !placeholder

Time Complexity?

##### !end-placeholder

##### !answer

/.+/

##### !end-answer

### !end-challenge

<!-- ======================= END CHALLENGE ======================= -->

<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->

### !challenge

* type: short-answer
* id: db5f2381-7288-442c-835e-0972ae72973a
* title: What is the space complexity of your solution?

> * points: 0

Reviewer: ```suggestion
* points: 1
```

* topics: Big-O, python, lists

##### !question

> What is the space complexity of your answer?

Reviewer: ```suggestion
What is the space complexity of your answer? Explain. Define your variable(s).
```
